### PR TITLE
[1162] Implement School Non JS search flow

### PR DESCRIPTION
**Pull request by darokel**
### Context

https://trello.com/c/JrVpNsaZ/1162-non-js-school-search-flow

### Changes proposed in this pull request

- Adds the non js school search routes, views, controllers etc
- Adds a service to perform the school search against the `gias_school` table
- Add new tsvector searchable field on `GiasSchool` to optimise searching of multiple `GiasSchool` attributes
- Migrations to add some indexes for the new searchable column

### Guidance to review

- Go to the providers schools tab and click add school
- Search for different attributes (try school and get loads of results or specific attributes like urns, postcode etc)

#### Caveats

- This PR only implements the search logic and flow. Populating the add school form with the school selected from the search will be carried out in a separate ticket after https://github.com/DFE-Digital/publish-teacher-training/pull/3466 is merged in.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Comment by darokel** [comment]
[Non JS school search flow](https://trello.com/c/JrVpNsaZ/1162-non-js-school-search-flow)

File: app/components/publish/schools/search_result_title_component.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          found_text,
+          query_text
+        ].compact.join(' ')
+      end
+
+      def results_text
+        return many_results_text if @results_count > results_limit
+        return "#{change_your_search_link}.".html_safe if @results_count.zero?
+
+        "#{change_your_search_link} if the school you're looking for is not listed.".html_safe
```
Comment (defong) [inline app/components/publish/schools/search_result_title_component.rb:26]:
```suggestion
        "#{change_your_search_link} if the school you’re looking for is not listed.".html_safe
```

**Review by defong — COMMENTED** [review]
> - [x] Inform data insights team due to database changes

:+1:

File: spec/forms/publish/schools/search_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+require 'rails_helper'
+
+module Publish
+  module Schools
+    describe SearchForm, type: :model do
+      subject { described_class.new }
+
+      it { is_expected.to validate_presence_of(:query) }
+
+      it { is_expected.to validate_length_of(:query).is_at_least(2) }
```
[the remark this answers is not in the document]
Comment (darokel) [reply spec/forms/publish/schools/search_form_spec.rb:12]: I'm pretty sure there was a good reason why we dropped it back to 2 characters but i just can't remember 😂

File: app/views/publish/providers/schools/index.html.erb [hunk 3]
```diff
@@ -35,6 +35,10 @@
         <%= render partial: "school", collection: @schools %>
       </tbody>
     </table>
-    <%= govuk_button_link_to "Add school", new_publish_provider_recruitment_cycle_school_path(@provider.provider_code) %>
+    <% if FeatureService.enabled?(:gias_search) %>
+      <%= govuk_button_link_to "Add school", search_publish_provider_recruitment_cycle_schools_path(@provider.provider_code) %>
```
Comment (goodviber) [inline app/views/publish/providers/schools/index.html.erb:39]: could we use a translation here?

File: app/views/publish/providers/schools/index.html.erb [hunk 4]
```diff
@@ -35,6 +35,10 @@
...
+    <% else %>
+      <%= govuk_button_link_to "Add school", new_publish_provider_recruitment_cycle_school_path(@provider.provider_code) %>
```
Comment (goodviber) [inline app/views/publish/providers/schools/index.html.erb:41]: could we use a translation here?

File: app/components/publish/schools/search_result_title_component.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      end
+
+      private
+
+      attr_reader :query, :results_limit, :results_count, :return_path
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/components/publish/schools/search_result_title_component.rb:31]: forgot about that - done

File: app/components/publish/schools/search_result_title_component.rb (hunk 1 above)
Comment (darokel) [reply app/components/publish/schools/search_result_title_component.rb]: done

File: spec/forms/publish/schools/search_form_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (darokel) [reply spec/forms/publish/schools/search_form_spec.rb:12]: Digging into the old trello card i think it was to make it consistent with with the JS autocomplete (the results)

File: app/views/publish/providers/schools/index.html.erb (hunk 4 above)
Comment (darokel) [reply app/views/publish/providers/schools/index.html.erb]: done and done

**Review by defong — APPROVED** [review]

**Review by simonwhatley — APPROVED** [review]

**Pull request merged**